pkg/filenotify/poller.go

```diff
@@ -146,12 +147,11 @@ func (w *filePoller) sendErr(e error, chClose <-chan struct{}) error {
 func (w *filePoller) watch(f *os.File, lastFi os.FileInfo, chClose chan struct{}) {
 	defer f.Close()
 	for {
-		time.Sleep(watchWaitTime)
 		select {
+		case <-time.After(watchWaitTime):
```

Comment:
You might have stray timers because of `After`, it's better to use `NewTimer` + `Stop` later I think.

Reply:
you mean Stop in defer ?

Reply:
Will the timer still exit? in here the watchWaitTime is 200ms

Reply:
Agree this should be fine.